pkg/agent/runner.go

```diff
@@ -511,6 +512,89 @@ func (r *Runner) handleVMResources(
 			reason = UpscaleRequested
 		case <-registeredScheduler.Recv():
 			reason = RegisteredScheduler
+		case <-vmInfoUpdated.Recv():
+			// Only actually do the update if something we care about changed:
+			newVMInfo := func() api.VmInfo {
+				r.status.lock.Lock()
+				defer r.status.lock.Unlock()
+				return r.status.vmInfo
+			}()
+
+			if !newVMInfo.ScalingEnabled {
+				// This shouldn't happen because any update to the VM object that has
+				// ScalingEnabled=false should get translated into a "deletion" so the runner stops.
+				// So we shoudln't get an "update" event, and if we do, something's gone very wrong.
+				panic("explicit VM update given but scaling is disabled")
+			}
+
+			// Update r.vm and r.lastApproved (see comment explaining why)
+			if changed := func() (changed bool) {
+				r.lock.Lock()
+				defer r.lock.Unlock()
+
+				if r.vm.Mem.SlotSize.Cmp(newVMInfo.Mem.SlotSize) != 0 {
+					panic("VM changed memory slot size")
```

Comment:
is this really a problem? seems like we could roll with it.

Reply:
Currently it would be a problem, yeah. But also it's not possible for VMs to change memory slot size at runtime, so if this happens something's probably gone horribly wrong somewhere else. Added a comment explaining that.